### DISCUSSION TITLE
Can this thread pool recursively add tasks?

### DISCUSSION THREAD

**Discussion by etorth** [ptsouchlos/thread-pool#72] [Q&A] [answered]
like following one:

https://github.com/vit-vit/CTPL/issues/36

**Comment by ptsouchlos** [comment, marked as answer]
In short, yes:

https://github.com/DeveloperPaul123/thread-pool/blob/f14f97d84d2b861b710564ce712275ca43acad60/test/source/thread_pool.cpp#L317-L374

But it's a bit more complicated than that. You can easily do things incorrectly this way. The thread pool tries to ensure that it doesn't get stuck, but it's still possible to block as mentioned here: https://github.com/DeveloperPaul123/thread-pool/issues/57#issue-2040318416

Also in the future, please open a discussion for questions instead of an issue.
